pypeit/pypeit.py

```diff
@@ -597,9 +594,24 @@ def reduce_exposure(self, frames, bg_frames=None, std_outfile=None):
                 all_specobjs_on_det = all_specobjs_objfind[all_specobjs_objfind.DET == detname]
             else:
                 all_specobjs_on_det = all_specobjs_objfind
+
```

Comment:
Can we just pass objFind_list[I] as an argument to self.extract_one() below and move this big of code from 598-610 to be inside extract_one. That would make the control flow in this main control loop of python a bit cleaner. 

In other words just write: 

all_spec2d[detname], tmp_sobjs = self.extract_one(frames, self.det, sciImg_list[i], objFind_list[I]
                                        final_global_sky, all_specobjs_on_det)

And put lines 598-610 into self.extract_one

Reply:
Done